pkg/base/addr_validation_test.go

```diff
@@ -64,12 +64,17 @@ func TestValidateAddrs(t *testing.T) {
 		t.Fatal("expected port resolution failure, got no error")
 	}
 	portExpectedErr := err.Error()
-	// For the host name resolution error we can reliably expect "no such host"
-	// below, but before we test anything we need to ensure we indeed have
-	// a reliably non-resolvable host name.
-	_, err = net.DefaultResolver.LookupIPAddr(context.Background(), "nonexistent.example.com")
-	if err == nil {
-		t.Fatal("expected host resolution failure, got no error")
+	// ValidateAddrs uses a net.DefaultResolver for host name resolution.
+	// Normally, when given a non-existent host name, we expect it to return a
+	// "no such host" error. However, it may return a different error if there is
+	// no network access. See
+	// https://github.com/cockroachdb/cockroach/issues/109052.
+	noSuchHostErr := func(host string) string {
+		_, err := net.DefaultResolver.LookupIPAddr(context.Background(), host)
+		if err == nil { // the supplied host is expected to be reliably non-resolvable
+			t.Fatal("expected host resolution failure, got no error")
+		}
+		return err.Error()
 	}
```

Comment:
have you even checked with @rickystewart whether this was a good fix?
I fear with this code that `noSuchHostErr("foo") == noSuchHostErr("bar")` and so the test below is testing something unrelated.

Reply:
> have you even checked with @rickystewart whether this was a good fix?

I haven't; @rickystewart thoughts?

> I fear with this code that `noSuchHostErr("foo") == noSuchHostErr("bar")` and so the test below is testing something unrelated.

`noSuchHostErr("foo") != noSuchHostErr("bar")` -- the host's name is printed in the error, which is actually why I had to pass the host name as a parameter. The error's something like "<hostname> no such host" IIRC.

Reply:
The fix seems to work inasmuch as it fixes the test. It works since we expect to get "some" error, whatever it is, and we just check whether it matches up to the expected error.

Note that this test will run in an environment without network access. To me this means the way the test is written is pretty misleading -- the IP lookup is going to fail in CI, whether the domain we're looking up is example.com or google.com or whatever. This also means that the "test" in CI is going to differ from the "test" as people run it on their local machines -- they'll be performing different `assert`s.

IMO it is more correct to [follow the advice I gave](https://github.com/cockroachdb/cockroach/issues/109052#issuecomment-1686600283) and mock the network lookup in a unit test: it is a *unit test*, so we shouldn't have a dependency on the network, and the test should be written such that we don't do useless network lookups that are guaranteed to fail.

For this, all I care about is whether the test is or is not failing, so I think it's fine as written.